### PR TITLE
FB8-56: Create a new mode SEMI_STRICT for slave_exec_mode

### PR DESCRIPTION
**Pull request by inikep**
JIRA: https://jira.percona.com/browse/FB8-56

Reference Patch: https://github.com/facebook/mysql-5.6/commit/9d97282

Summary:
Create a new mode SEMI_STRICT which ignores HA_ERR_RECORD_CHANGED and HA_ERR_KEY_NOT_FOUND errors.
A new status counter will track the usage of this new mode. Logging a warning/error message may cause error log to get flooded with same error message, so I decided against it.

Test Plan: mtr

### CODE REVIEW

File: mysql-test/suite/rpl/t/rpl_slave_exec_mode_semi_strict.test [hunk 1]
```diff
@@ -0,0 +1,19 @@
+source include/have_binlog_format_row.inc;
+source include/master-slave.inc;
+connection master;
+
+create table t1 (a int primary key, b int) engine=Innodb;
+insert into t1 values(1, 2);
+
+sync_slave_with_master;
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_slave_exec_mode_semi_strict.test]: use `--source include/sync_slave_sql_with_master.inc` instead

File: mysql-test/suite/rpl/t/rpl_slave_exec_mode_semi_strict.test [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+delete from t1 where a=1;
+
+connection master;
+delete from t1 where a=1;
+sync_slave_with_master;
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_slave_exec_mode_semi_strict.test]: use `--source include/sync_slave_sql_with_master.inc` instead

File: mysql-test/suite/rpl/t/rpl_slave_exec_mode_semi_strict.test [hunk 3]
```diff
@@ -0,0 +1,19 @@
+source include/have_binlog_format_row.inc;
+source include/master-slave.inc;
+connection master;
+
+create table t1 (a int primary key, b int) engine=Innodb;
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_slave_exec_mode_semi_strict.test]: UPPER case for SQL keywords

File: mysql-test/suite/rpl/t/rpl_slave_exec_mode_semi_strict.test [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+insert into t1 values(1, 2);
+
+sync_slave_with_master;
+delete from t1 where a=1;
+
+connection master;
+delete from t1 where a=1;
+sync_slave_with_master;
+show global status like 'Rbr_unsafe_queries';
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_slave_exec_mode_semi_strict.test]: Make sure this test passes with `--repeat=2`. If not, save initial value `Rbr_unsafe_queries`, do delete and then assert that `Rbr_unsafe_queries` changed by 1.

File: sql/mysqld.cc [hunk 5]
```diff
@@ -8687,6 +8689,8 @@ SHOW_VAR status_vars[] = {
     {"Queries", (char *)&show_queries, SHOW_FUNC, SHOW_SCOPE_ALL},
     {"Questions", (char *)offsetof(System_status_var, questions),
      SHOW_LONGLONG_STATUS, SHOW_SCOPE_ALL},
+    {"Rbr_unsafe_queries", (char *)&rbr_unsafe_queries, SHOW_LONGLONG,
```
Comment (percona-ysorokin) [inline sql/mysqld.cc:8692]: Does this new status var need to be reflected in `perfschema.show_sanity` MTR test case?

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: sql/mysqld.cc (hunk 5 above)
Comment (inikep) [reply sql/mysqld.cc:8692]: There is no `perfschema.show_sanity` in 8.0

**Comment by inikep** [comment]
PR was updated with requested changes

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Review by hermanlee — APPROVED** [review]

**Pull request closed without merging**